lib/shopify_cli/theme/dev_server/proxy_param_builder.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require "cgi"
+
+module ShopifyCLI
+  module Theme
+    module DevServer
+      class ProxyParamBuilder
+        def build
+          # Core doesn't support replace_templates
+          return {} if core?(current_path)
+
+          (syncer_templates + request_templates)
+            .select { |file| file.liquid? || file.json? }
+            .uniq(&:relative_path)
+            .map { |file| as_param(file) }
+            .to_h
+        end
+
+        def with_core_endpoints(core_endpoints)
+          @core_endpoints = core_endpoints
+          self
+        end
+
+        def with_syncer(syncer)
+          @syncer = syncer
+          self
+        end
+
+        def with_rack_env(rack_env)
+          @rack_env = rack_env
+          self
+        end
+
+        def with_theme(theme)
+          @theme = theme
+          self
+        end
+
+        private
+
+        def as_param(file)
+          ["replace_templates[#{file.relative_path}]", file.read]
+        end
+
+        def syncer_templates
+          @syncer&.pending_updates || []
+        end
+
+        def request_templates
+          cookie_sections
+            .map { |section| @theme[section] unless @theme.nil? }
+            .compact
+        end
+
+        def cookie_sections
+          CGI::Cookie.parse(cookie)["hot_reload_files"].join.split(",") || []
```

Comment:
Does this account for the `shopify theme serve` command using the `hot_reload_sections` cookie?

Reply:
Good catch, @jessetuglu! I've renamed everywhere to the `hot_reload_files` (in the 🎩, I didn't spot any issue probably because of the syncer [redundancy](https://github.com/Shopify/shopify-cli/blob/proxy-refactoring/lib/shopify_cli/theme/dev_server/proxy_param_builder.rb#L13)).